src/main/java/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package strategy;
+
+import java.util.Map;
+import model.FruitTransaction;
+import service.OperationHandler;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<FruitTransaction.Operation,
+            OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(FruitTransaction.Operation operation) {
```

Comment:
It's good practice to validate the input parameters to prevent potential issues. Consider checking if the 'operation' is null or if the 'operationHandlers' map contains a handler for the given operation before returning it.